core/ast/src/source_text.rs

```diff
@@ -0,0 +1,72 @@
+use crate::{LinearPosition, LinearSpan};
+
+/// Source text.
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
+#[derive(Clone, Debug)]
+pub struct SourceText {
+    source_text: Vec<u16>,
+}
+
+impl SourceText {
+    /// Constructs a new, empty `SourceText` with at least the specified capacity.
+    #[must_use]
+    pub fn with_capacity(capacity: usize) -> Self {
+        Self {
+            source_text: Vec::with_capacity(capacity),
+        }
+    }
+    /// Get current `LinearPosition`.
+    #[must_use]
+    pub fn cur_linear_position(&self) -> LinearPosition {
```

Comment:
Just for readability it would be great if you could leave an empty line between functions so the file does not look so cluttered.